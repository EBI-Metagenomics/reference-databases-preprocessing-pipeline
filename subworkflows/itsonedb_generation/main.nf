
include { UNCOMPRESS_FILE } from '../../modules/local/uncompress_file/main.nf'
include { GET_TAX_LINEAGE } from '../../modules/local/get_tax_lineage/main.nf'
include { ITSONEDB_REFORMAT } from '../../modules/local/itsonedb_reformat/main.nf'
include { GENERATE_ITSONEDB_TAX } from '../../modules/local/generate_itsonedb_tax/main.nf'
include { ITSONEDB_COLUMN_REPLACEMENT } from '../../modules/local/itsonedb_column_replacement/main.nf'
include { CLEAN_FASTA } from '../../modules/local/clean_fasta/main.nf'
include { MAKE_OTU_FILE } from '../../modules/local/make_otu_file/main.nf'
include { GENERATE_MSCLUSTER } from '../../modules/local/generate_mscluster/main.nf'


workflow ITSONEBD_GENERATION {

    main:

        dummy_fasta = file(params.dummy_fasta)
        itsonedb_fasta = file(params.itsonedb_download_fasta)
        itsonedb_download_taxdump = file(params.itsonedb_download_taxdump)

        UNCOMPRESS_FILE(
            itsonedb_fasta,
            "itsonedb.fasta"
        )

        GET_TAX_LINEAGE(
            UNCOMPRESS_FILE.out.uncmp_file,
            itsonedb_download_taxdump,
            params.itsonedb_version,
            params.itsonedb_label
        )

        ITSONEDB_REFORMAT(
            GET_TAX_LINEAGE.out.tax_lineage,
            params.itsonedb_version,
            params.itsonedb_label
        )

        GENERATE_ITSONEDB_TAX(
            UNCOMPRESS_FILE.out.uncmp_file,
            ITSONEDB_REFORMAT.out.uplift,
            params.itsonedb_version,
            params.itsonedb_label
        )

        ITSONEDB_COLUMN_REPLACEMENT(
            GENERATE_ITSONEDB_TAX.out.uplift_final,
            ITSONEDB_REFORMAT.out.taxid,
            params.itsonedb_tax_header,
            params.itsonedb_version,
            params.itsonedb_label
        )

        CLEAN_FASTA(
            UNCOMPRESS_FILE.out.uncmp_file,
            ITSONEDB_COLUMN_REPLACEMENT.out.tax,
            params.itsonedb_version,
            params.itsonedb_label
        )

        MAKE_OTU_FILE(
            ITSONEDB_COLUMN_REPLACEMENT.out.tax,
            ITSONEDB_COLUMN_REPLACEMENT.out.taxid,
            params.itsonedb_version,
            params.itsonedb_label
        )

        GENERATE_MSCLUSTER(
            dummy_fasta,
            CLEAN_FASTA.out.cleaned_fasta,
            ITSONEDB_COLUMN_REPLACEMENT.out.tax,
            params.itsonedb_version,
            params.itsonedb_label
        )

    emit:
        fasta = CLEAN_FASTA.out.cleaned_fasta
        tax = ITSONEDB_COLUMN_REPLACEMENT.out.tax
        otu = MAKE_OTU_FILE.out.otu

}