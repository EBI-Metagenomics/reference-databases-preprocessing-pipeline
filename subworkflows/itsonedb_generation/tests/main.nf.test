nextflow_workflow {

    name "Test Workflow ITSONEBD_GENERATION"
    script "subworkflows/itsonedb_generation/main.nf"
    workflow "ITSONEBD_GENERATION"
    tag "subworkflows"
    tag "itsonedb_generation"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
                itsonedb_version = "1.141"
                itsonedb_label = "ITSone"
                itsonedb_download_fasta = "${projectDir}/subworkflows/itsonedb_generation/tests/data/test.fasta.gz"
                itsonedb_download_taxdump = "${projectDir}/subworkflows/itsonedb_generation/tests/data/sdwca.zip"
                itsonedb_tax_header = "${projectDir}/subworkflows/itsonedb_generation/tests/data/itsonedb_tax_header.txt"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.fasta).match("fasta") },
                { assert snapshot(workflow.out.tax).match("tax") },
                { assert snapshot(workflow.out.otu).match("otu") }
            )
        }

    }

}
