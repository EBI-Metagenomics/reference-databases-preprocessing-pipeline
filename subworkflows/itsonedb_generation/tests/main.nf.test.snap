{
    "otu": {
        "content": [
            [
                "ITSone.otu:md5,0dca086b2cd96e90b89ddba6a7b3b9b2"
            ]
        ],
        "timestamp": "2024-01-04T11:53:48.036734317"
    },
    "tax": {
        "content": [
            [
                "ITSone-tax.txt:md5,c9dcddf664113dc4ab8ff9e500d9ccd0"
            ]
        ],
        "timestamp": "2024-01-04T11:53:47.946471005"
    },
    "fasta": {
        "content": [
            [
                "ITSone.fasta:md5,58fbd1ee1e83fbe4c157d1bcdeceba4f"
            ]
        ],
        "timestamp": "2024-01-04T11:53:47.874652776"
    }
}